python/pylibcugraph/pylibcugraph/heterogeneous_neighbor_sample.pyx

```diff
@@ -0,0 +1,511 @@
+    cugraph_heterogeneous_neighbor_sample,
+    cugraph_sample_heterogeneous_fan_out_t,
+    cugraph_create_heterogeneous_fan_out,
+    cugraph_heterogeneous_fan_out_free,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+    assert_AI_type,
+    get_c_type_from_numpy_type,
+)
+from pylibcugraph.internal_types.sampling_result cimport (
+    SamplingResult,
+)
+from pylibcugraph._cugraph_c.random cimport (
+    cugraph_rng_state_t
+)
+from pylibcugraph.random cimport (
+    CuGraphRandomState
+)
+import warnings
+
+# TODO accept cupy/numpy random state in addition to raw seed.
+def heterogeneous_neighbor_sample(ResourceHandle resource_handle,
+                    _GPUGraph input_graph,
+                    start_list,
+                    h_fan_out,
+                    bool_t is_biased,
+                    bool_t with_replacement,
+                    bool_t do_expensive_check,
+                    with_edge_properties=False,
+                    batch_id_list=None,
+                    label_list=None,
+                    label_to_output_comm_rank=None,
+                    label_offsets=None,
+                    prior_sources_behavior=None,
+                    deduplicate_sources=False,
+                    return_hops=False,
+                    renumber=False,
+                    retain_seeds=False,
+                    compression='COO',
+                    compress_per_hop=False,
+                    random_state=None,
+                    return_dict=False,):
+    """
+    # FIXME: Deprecate uniform_neighbor_sample
+    Does both uniform and biased neighborhood sampling, which samples nodes from
+    a graph based on the current node's neighbors, with a corresponding fan_out
+    value at each hop.
+
+    # FIXME: biased neighbor sampling is not yet exposed to PLC.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    input_graph : SGGraph or MGGraph
+        The input graph, for either Single or Multi-GPU operations.
+
+    start_list: device array type
+        Device array containing the list of starting vertices for sampling.
+
+    edge_biases: FIXE: update this - Create edge_biases of type
+    'cugraph_edge_property_view_t' - edge biases not yet supported.
+
+    h_fan_out: tuple of numpy array type
+        Device array containing the branching out (fan-out) degrees per
+        starting vertex for each hop level in CSR style format. The first
+        element of the tuple is the offset array per edge type id and the second
+        element corresponds to the fan_out values.
+        The sampling method can use different fan_out values for each edge type.
+    
+    is_biased: bool
```

Comment:
We made the C++ parameter obsolete be separating the uniform and biased methods.  We should mirror this in PLC.